vpr/src/place/analytic_placer.h

```diff
@@ -0,0 +1,160 @@
+#ifndef VPR_ANALYTIC_PLACEMENT_H
+#define VPR_ANALYTIC_PLACEMENT_H
+
+#include "vpr_context.h"
+#include "timing_place.h"
+#include "PlacementDelayCalculator.h"
+
+typedef signed long int wirelen_t;
+
+template<typename T>
+struct EquationSystem;
+
+class AnalyticPlacer {
+  public:
+    AnalyticPlacer(VprContext& ctx);
+
+    void ap_place();
+
+  private:
+    // for CutSpreader to access legal_pos, block_locs, etc.
+    friend class CutSpreader;
+
+    // TODO: PlacerHeapCfg should be externally configured & supplied
+    struct PlacerHeapCfg {
+        float alpha, beta;
+        int criticalityExponent;
+        int timingWeight;
+        float solverTolerance;
+        bool placeAllAtOnce;
+        int spread_scale_x, spread_scale_y;
+    };
+
+    PlacerHeapCfg tmpCfg;
+
+    int max_x = 0, max_y = 0; //size of the grid
```

Comment:
Are max_x and max_y the same as the grid.width and height variables?  If so, why make a copy?  I think it would be more clear to just work with the originals.